sql/src/test/java/io/crate/integrationtests/InsertIntoIntegrationTest.java

```diff
@@ -1458,4 +1458,133 @@ public void testInsertDefaultExpressions() {
                "5| cr8\n")
         );
     }
+
+    @Test
+    public void test_insert_with_id_in_returning_clause() {
+        execute("create table t (" +
+                " id int," +
+                " owner text default 'crate'" +
+                ") with (number_of_replicas=0)");
+
+        execute("insert into t (id) values (?) returning id",
+                new Object[]{1});
+
+        assertThat(response.rowCount(), is(1L));
+        assertThat(response.cols()[0], is("id"));
+        assertThat(response.rows()[0][0], is(1));
+
+    }
+
+    @Test
+    public void test_insert_with_multiple_values_in_returning_clause() {
+        execute("create table t (" +
+                " id int," +
+                " owner text default 'crate'" +
+                ") with (number_of_replicas=0)");
+
+        execute("insert into t (id) values (?) returning id, owner",
+                new Object[]{1});
+
+        assertThat(response.rowCount(), is(1L));
+        assertThat(response.cols()[0], is("id"));
+        assertThat(response.cols()[1], is("owner"));
+        assertThat(response.rows()[0][0], is(1));
+        assertThat(response.rows()[0][1], is("crate"));
+
+    }
+
+    @Test
+    public void test_insert_with_function_in_returning_clause() {
+        execute("create table t (" +
+                " id int," +
+                " owner text default 'crate'" +
+                ") with (number_of_replicas=0)");
+
+        execute("insert into t (id) values (?) returning id + 1 as bar",
+                new Object[]{1});
+
+        assertThat(response.rowCount(), is(1L));
+        assertThat(response.cols()[0], is("bar"));
+        assertThat(response.rows()[0][0], is(2));
+
+    }
+
+
+    @Test
+    public void test_insert_with_seq_no_in_returning_clause() {
+        execute("create table t (" +
+                " id int," +
+                " owner text default 'crate'" +
+                ") with (number_of_replicas=0)");
+
+        execute("insert into t (id) values (?) returning _seq_no as seq",
+                new Object[]{1});
+
+        assertThat(response.rowCount(), is(1L));
+        assertThat(response.cols()[0], is("seq"));
+        assertThat(response.rows()[0][0], is(0L));
+
+    }
+
+    @Test
+    public void test_insert_with_owner_renamed_in_returning_clause() {
+        execute("create table t (" +
+                " id int," +
+                " owner text default 'crate'" +
+                ") with (number_of_replicas=0)");
+
+        execute("insert into t (id) values (?) returning owner as name",
+                new Object[]{1});
+
+        assertThat(response.rowCount(), is(1L));
+        assertThat(response.cols()[0], is("name"));
+        assertThat(response.rows()[0][0], is("crate"));
+
+    }
+
+        public void test_insert_from_subquery_with_id_field_in_returning_clause() {
```

Comment:
format: indent issue (also `@Test` annotation missing)

Reply:
Good catch, thank you!